flag-app/flag-app-project/src/CountriesContextProvider.tsx

```diff
@@ -4,12 +4,21 @@ import { useApiData } from "./hooks";
 
 const CountriesContextProvider: FC<PropsWithChildren> = ({ children }) => {
   const [searchInput, setSearchInput] = useState("");
-  const { data: countries, error, isLoading } = useApiData(searchInput);
+  const [filterValue, setFilterValue] = useState("");
+  const {
+    data: countries,
+    error,
+    isLoading,
+  } = useApiData(searchInput, filterValue || "");
```

Comment:
`filterValue || ""` will never trigger the other end, so only `filterValue` is enough. `searchInput` and `filterValue` are both of type `string`